src/api/procedures/executeManualInstruction.ts

```diff
@@ -0,0 +1,174 @@
+import { PolymeshPrimitivesIdentityIdPortfolioId } from '@polkadot/types/lookup';
+import BigNumber from 'bignumber.js';
+import P from 'bluebird';
+
+import { assertInstructionValidForManualExecution } from '~/api/procedures/utils';
+import { Instruction, PolymeshError, Procedure } from '~/internal';
+import {
+  DefaultPortfolio,
+  ErrorCode,
+  ExecuteManualInstructionParams,
+  InstructionDetails,
+  Leg,
+  NumberedPortfolio,
+  TxTags,
+} from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+import {
+  bigNumberToU32,
+  bigNumberToU64,
+  portfolioIdToMeshPortfolioId,
+  portfolioLikeToPortfolioId,
+  u64ToBigNumber,
+} from '~/utils/conversion';
+
+export interface Storage {
+  portfolios: (DefaultPortfolio | NumberedPortfolio)[];
+  totalLegAmount: BigNumber;
+  instructionDetails: InstructionDetails;
+  signer: string;
```

Comment:
```suggestion
  signerDid: string;
```

Usually a "signer" is more associated to an address or key. I think Did makes it clearer what it is